### PR TITLE
Basic: use `.standardizingPath` instead of `realpath`

### PR DESCRIPTION
**Pull request by compnerd**
This Foundation extension provides the same behaviour but is portable
across different platforms.

### CODE REVIEW

**Comment by compnerd** [comment]
@swift-ci please smoke test

**Comment by compnerd** [comment]
@swift-ci please smoke test

File: Sources/Basic/PathShims.swift [hunk 1]
```diff
@@ -17,22 +17,11 @@
...
-    // FIXME: We can't use FileManager's destinationOfSymbolicLink because
-    // that implements readlink and not realpath.
-    if let resultPtr = SPMLibc.realpath(pathStr, nil) {
-        let result = String(cString: resultPtr)
-        // FIXME: We should measure if it's really more efficient to compare the strings first.
-        return result == pathStr ? path : AbsolutePath(result)
-    }
-
-    return path
+    return AbsolutePath(path.pathString.resolvingSymlinksInPath)
```
Comment (abertelrud) [inline Sources/Basic/PathShims.swift]: I understand and support the motivation here, but I have some concerns about performance.  For example, I think that the swift-corelibs-foundation implementation implements this by creating multiple URLs and breaking it into components.  Would it be possible to conditionally keep the optimized implementation and use the more generic one only for platforms that don't support `realpath()`?

We should of course measure to see whether there's really a difference, but shouldn't make this change unless we know it doesn't slow down the current platforms.

Comment (karwa) [reply Sources/Basic/PathShims.swift]: Just a curious onlooker here, but this seems like a strange criticism: the Foundation API does exactly what this one is supposed to do, and if there are more optimal ways to get the result, that should be a Foundation-level optimisation which would benefit everybody.

**Comment by compnerd** [comment]
Obsoleted by apple/swift-tools-support-core#114

**Pull request closed without merging**